### PR TITLE
[ROCm] Adding ROCm support for the relu op

### PR DESCRIPTION
**Pull request by deven-amd**
This PR adds ROCm support for the relu op

The changes in this PR are trivial. please review and merge. thanks.

-------------------------------------------------

@tatianashp @whchung @chsigg

### CODE REVIEW

**Review by whchung — DISMISSED** [review]

File: tensorflow/core/kernels/relu_op.cc [hunk 1]
```diff
@@ -143,11 +143,13 @@ namespace functor {
       typename TTypes<T>::Tensor backprops);                                   \
   extern template struct SeluGrad<GPUDevice, T>;
 
+#if GOOGLE_CUDA
```
Comment (chsigg) [inline tensorflow/core/kernels/relu_op.cc]: Could you add a comment that this is currently not supported on ROCm?

Comment (deven-amd) [reply tensorflow/core/kernels/relu_op.cc]: done

**Review by whchung — APPROVED** [review]

**Comment by deven-amd** [comment]
@rthadur 

this is the 4th of 4 PRs that seem to be stuck in the merge pipeline. 
Please let me know if there is anything that needs to be done on my end.

thanks

deven

**Comment by rthadur** [comment]
@deven-amd thank you for your patience waiting for @chsigg review in order to pull this in.

**Comment by dagamayank** [comment]
@chsigg  gentle ping.

**Comment by dagamayank** [comment]
@chsigg Can you please review this PR?

File: tensorflow/core/kernels/relu_op.cc [hunk 2]
```diff
@@ -188,6 +191,7 @@ TF_CALL_GPU_NUMBER_TYPES(DECLARE_GPU_SPEC);
 TF_CALL_GPU_NUMBER_TYPES(REGISTER_GPU_KERNELS);
 #undef REGISTER_GPU_KERNELS
 
+#if GOOGLE_CUDA
```
Comment (chsigg) [inline tensorflow/core/kernels/relu_op.cc]: Doesn't this disable the CPU implementation?

Comment (deven-amd) [reply tensorflow/core/kernels/relu_op.cc]: It does not seem so.

This region of code (lines 77 - 217) is within the `#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM` block

So this `#if GOOGLE_CUDA` will disable only the ROCm implementation.

Comment (chsigg) [reply tensorflow/core/kernels/relu_op.cc]: Yeah, your are right. Thanks for the explanation.

**Review by chsigg — APPROVED** [review]

**Pull request merged**